### ISSUE TITLE
file_is_svg reads the whole file in memory, causing an OOM error for big files

### ISSUE THREAD

**Issue by Znuff** [projectsend/projectsend#918]
Specifically: https://github.com/projectsend/projectsend/blob/master/includes/functions.php#L1140-L1152

For large files that are Larger than PHP's `memory_limit`, this will cause an error:

```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 3031609376 bytes) [...]
```

Increasing the memory to absurd amounts is not a solution.

**Comment by ignacionelson** [comment]
I see. I guess a temp patch would be to limit the size of testable files. Simply trusting the file is not option.
Files larger than the max setting would not be approved as svg for previews.

**Comment by Znuff** [comment]
I think a better approach is to https://www.php.net/manual/en/book.fileinfo.php and check if the filetype is actually SVG.

**Comment by ignacionelson** [comment]
Fixed on 0e32bbb
Thanks!
